packages/malloc/src/pool.ts

```diff
@@ -107,41 +173,40 @@ export class MemPool implements IMemPool {
                 block.next = this._used;
                 this._used = block;
                 if (isTop) {
-                    block.size = size;
-                    this.top = block.addr + size;
+                    block.size = memorySize;
+                    this.top = block.addr + memorySize;
                 } else if (this.doSplit) {
-                    const excess = block.size - size;
+                    const excess = block.size - memorySize;
                     if (excess >= this.minSplit) {
-                        block.size = size;
-                        this.insert({
-                            addr: block.addr + size,
-                            size: excess,
-                            next: null
-                        });
+                        block.size = memorySize;
+                        const newBlock = new MemBlockWrapper(this.dataView, block.addr + memorySize)
```

Comment:
Ahh... so `dataView` is used here too. In that case it'd probably be best to just add a `this.u32` view of the whole array buffer, similarly to the already existing `this.u8` view (created in the pool's constructor) and then just pass this to `new MemBlockWrapper` and also make sure all array indices using that view are based on `index >> 2`...